notebooks/HASP/COSLifetimePositions/COSLifetimePositions.ipynb

```diff
@@ -0,0 +1,1844 @@
+    "* `scipy.optimize.curve_fit` to fit a function to a spectrum\n",
+    "* `shutil` to perform directory and file operations\n",
+    "* `specutils.Spectrum1d` to create Spectrum1d objects\n",
+    "\n",
+    "We recommend creating a HASP-specific `conda` environment. You can check out our [Setup.ipynb](https://github.com/spacetelescope/hst_notebooks/tree/main/notebooks/HASP/Setup) notebook to create such an environment. Alternatively, you can also download the required dependencies to run this notebook with the terminal command: \n",
+    "\n",
+    "`pip install -r requirements.txt`\n",
+    "\n",
+    "Let's import all of the packages that we will use in this notebook by running the next cell:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "4334aa1c",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The history saving thread hit an unexpected error (OperationalError('attempt to write a readonly database')).History will not be written to the database.\n"
+     ]
+    }
+   ],
+   "source": [
+    "%matplotlib inline\n",
+    "\n",
+    "import glob\n",
+    "import matplotlib\n",
+    "import numpy as np\n",
+    "import os\n",
+    "import shutil\n",
+    "\n",
+    "from astropy import units as u\n",
+    "from astropy.convolution import convolve\n",
+    "from astropy.io import ascii, fits\n",
+    "from astropy.visualization import quantity_support\n",
+    "from astroquery.mast import Observations\n",
+    "from matplotlib import pyplot as plt\n",
+    "from pathlib import Path\n",
+    "from scipy.optimize import curve_fit\n",
+    "from specutils import Spectrum1D\n",
+    "\n",
+    "# Specify plot parameters.\n",
+    "quantity_support()  # to put units on the axes below \n",
+    "matplotlib.rcParams['figure.figsize'] = [15, 5]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6dd2dbfd",
+   "metadata": {},
+   "source": [
+    "<a id = lsfdownload></a>\n",
+    "## 1. Download the COS Line-Spread Functions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6bdca358",
+   "metadata": {},
+   "source": [
+    "While it is possible to download the various COS line-spread functions via Python, it is easier to retrieve them by hand.  From the [COS homepage](https://www.stsci.edu/hst/instrumentation/cos/performance/spectral-resolution), follow the link to the [COS Line Spread Functions](https://www.stsci.edu/hst/instrumentation/cos/performance/spectral-resolution).  Scroll down to the tables containing the various LSFs.  For each lifetime position (1 to 5), retrieve the LSF for grating G130M and central wavelength 1291.  Be sure to get the \"Tabulated Theoretical Line Spread Functions\" on the left side of the page, NOT the \"Cross Dispersion\" functions on the right.  The files should have names like \"aa_LSFTable_G130M_1291_LP1_cn.dat.\"  Store them in a subdirectory called \"LSF\".\n",
```

Comment:
this needs to be done programmatically otherwise CI will fail on this notebook. Is there another COS notebook that uses these that you can use as an example?